### PR TITLE
fix: even in nvim, python2 and python3 providers may not be provided both

### PR DESCRIPTION
**Pull request by laishulu**
#672

### CODE REVIEW

File: autoload/jedi.vim [hunk 1]
```diff
@@ -68,8 +68,7 @@ function! s:init_python() abort
     endif
 
     " Handle "auto" version.
-    if has('nvim') || (has('python') && has('python3'))
-        " Neovim usually has both python providers. Skipping the `has` check
+    if (has('python') && has('python3'))
```
Comment (blueyed) [inline autoload/jedi.vim]: The `has('nvim')` is meant to not trigger the loading of both Python hosts in Neovim.

Comment (blueyed) [inline autoload/jedi.vim]: I think you should set `let g:jedi#force_py_version = 3` instead.

Comment (blueyed) [inline autoload/jedi.vim]: I think we could improve it here, but I would then expect `let g:loaded_python_provider = 1` for it to be available.
So in this case you should set `let g:loaded_python_provider = 0`, which also would skip loading it, but then can be considered to mean "do not use it".

**Comment by blueyed** [comment]
```diff
diff --git i/autoload/jedi.vim w/autoload/jedi.vim
index a101836..762bddf 100644
--- i/autoload/jedi.vim
+++ w/autoload/jedi.vim
@@ -69,7 +69,9 @@ function! s:init_python() abort
     endif
 
     " Handle "auto" version.
-    if has('nvim') || (has('python') && has('python3'))
+    if (has('nvim') && get(g:, 'loaded_python_provider', 1)
+          \   && get(g:, 'loaded_python3_provider', 1))
+          \ || has('python') && has('python3')
         " Neovim usually has both python providers. Skipping the `has` check
         " avoids starting both of them.
 ```

**Comment by blueyed** [comment]
Can you check if https://github.com/davidhalter/jedi-vim/pull/697 works for you? (especially https://github.com/davidhalter/jedi-vim/pull/697/commits/a6bbb5d5c794cba0414954480f4cb12732aab8ec)

**Comment by davidhalter** [comment]
Python 2 has been removed, I think this PR is therefore obsolete.

**Pull request closed without merging**